p2p/peer_tracker.go

```diff
@@ -179,6 +206,17 @@ func (p *peerTracker) gc() {
 					delete(p.trackedPeers, id)
 				}
 			}
+
+			peerlist := make([]peer.AddrInfo, 0, len(p.trackedPeers))
```

Comment:
for now and for the purpose of this implementation, we can just store all tracked peers but eventually we could be more sophisticated with determination.

Reply:
Agreed. Actually we should fish out the trusted peers from the store because if we don't, in the `HEAD` request, when `subjectiveInit` is false, we would end up with a peerset containing same peers twice and thus we would send them `HEAD` requests twice. Thoughts?